src/App.tsx

```diff
@@ -1,23 +1,41 @@
 import { RouterProvider, createBrowserRouter } from "react-router-dom";
 import DashboardScreen from "./components/screens/dashboard/dashboard-screen";
 import ErrorPage from "./components/screens/error-page";
+import AuthenticationProvider from "./components/providers/authentication-provider";
+import { useState } from "react";
+import { type KeycloakProfile, type KeycloakTokenParsed } from "keycloak-js";
 
-const router = createBrowserRouter([
-  {
-    path: "/",
-    element: <DashboardScreen />,
-    errorElement: <ErrorPage />
-  }
-]);
+export interface Auth {
+  token: KeycloakTokenParsed | undefined
+  tokenRaw: string | undefined
+  logout: () => void
+}
 
 /**
  * Application component
  *
  */
 function App () {
+  const [auth, setAuth] = useState<Auth>();
+  const [userProfile, setUserProfile] = useState<KeycloakProfile | undefined>(undefined);
```

Comment:
```suggestion
  const [userProfile, setUserProfile] = useState<KeycloakProfile>();
```
`undefined` value doesn't need to be explicitly typed nor set.

Reply:
Fixed